### PR TITLE
[DOC] Added on modifier example to the input component

### PR DESCRIPTION
**Pull request by stevetyler**
https://github.com/emberjs/ember.js/issues/19201

### CODE REVIEW

File: packages/@ember/-internals/glimmer/lib/components/input.ts [hunk 1]
```diff
@@ -38,7 +38,13 @@ import InternalComponent from './internal';
 
   ### Actions
 
-  The `Input` component takes a number of arguments with callbacks that are invoked in response to
+  There are multiple ways to pass actions to the input component. Starting with Ember Octane, we recommend using the {{on}} modifier to call an action on specific events such as the input event.
```
Comment (jenweber) [inline packages/@ember/-internals/glimmer/lib/components/input.ts:41]:
```suggestion
  There are multiple ways to pass actions to the input component. Starting with Ember Octane, we recommend using the `{{on}}` modifier to call an action on specific events such as the input event.
```

**Review by jenweber — APPROVED** [review]
Looks great! Thank you!

**Comment by jenweber** [comment]
@locks I tagged you since I think you know how/when to merge things in this repo. If not, can you tag another reviewer?

**Comment by jenweber** [comment]
@chancancode this PR aligns the API docs with what we landed on for the Guides https://github.com/ember-learn/guides-source/pull/1546. If you could review & merge if it seems good, that would be great! Thank you!

File: packages/@ember/-internals/glimmer/lib/components/input.ts [hunk 2]
```diff
@@ -38,7 +38,13 @@ import InternalComponent from './internal';
 
   ### Actions
 
-  The `Input` component takes a number of arguments with callbacks that are invoked in response to
+  There are multiple ways to pass actions to the input component. Starting with Ember Octane, we recommend using the `{{on}}` modifier to call an action on specific events such as the input event.
```
Comment (chancancode) [inline packages/@ember/-internals/glimmer/lib/components/input.ts:41]:
```suggestion
  The recommended way of handling DOM events is to use the `{{on}}` modifier. For example, to specify a callback for the `input` event:
```

File: packages/@ember/-internals/glimmer/lib/components/input.ts [hunk 3]
```diff
@@ -38,7 +38,13 @@ import InternalComponent from './internal';
 
   ### Actions
```
Comment (chancancode) [inline packages/@ember/-internals/glimmer/lib/components/input.ts:39]:
```suggestion
  ### Events
```

File: packages/@ember/-internals/glimmer/lib/components/input.ts [hunk 4]
```diff
@@ -38,7 +38,13 @@ import InternalComponent from './internal';
...
 
-  The `Input` component takes a number of arguments with callbacks that are invoked in response to
+  There are multiple ways to pass actions to the input component. Starting with Ember Octane, we recommend using the `{{on}}` modifier to call an action on specific events such as the input event.
+
+  ```handlebars
+  <Input @value={{this.name}} {{on "input" this.validateName}} />
+  ```
+
+  The `Input` component also takes a number of arguments with callbacks that are invoked in response to
   user events.
```
Comment (chancancode) [inline packages/@ember/-internals/glimmer/lib/components/input.ts:42]:
```suggestion
  other events.
```

File: packages/@ember/-internals/glimmer/lib/components/input.ts (hunk 2 above)
Comment (chancancode) [reply packages/@ember/-internals/glimmer/lib/components/input.ts:41]: Can we cross link to the documentation for the on modifier if that's available?

File: packages/@ember/-internals/glimmer/lib/components/input.ts [hunk 5]
```diff
@@ -38,7 +38,13 @@ import InternalComponent from './internal';
...
 
   * `enter`
```
Comment (chancancode) [inline packages/@ember/-internals/glimmer/lib/components/input.ts:44]:
```suggestion
  * `enter` (synthetic event)
```

IMO we should do this for this for all the synthetic events and at the end of the list mention what they are (something along the lines of "Some of these do not correspond to a DOM event. Instead, they are synthetic/logical events that are fired by the `Input` component when a high-level user action is take (such as pressing the Enter key).")

**Review by chancancode — COMMENTED** [review]
What do you think?

**Comment by mansona** [comment]
Hey @stevetyler 👋  have you seen @chancancode 's suggestions? let us know if you need any help moving this forward 👍

**Comment by locks** [comment]
Hey @stevetyler :) Do you have some time or do you need some help with the remaining work?

**Comment by bertdeblock** [comment]
#19314 was made (and merged) to address the same issue. Is this PR still needed?

**Comment by rwjblue** [comment]
kk, good point @bertdeblock thanks for the reminder!

Closing in favor of https://github.com/emberjs/ember.js/pull/19314 (thanks for everyones hard work here!!).

**Pull request closed without merging**